.emacs.d/lisp/visual/line-numbers-c.el

```diff
@@ -0,0 +1,44 @@
+(defun kotct/line-numbers--set-up-linum ()
+  "Sets up `linum-mode' to run globally except in a few cases."
+  (progn
+    ;; enable linum globally
+    (global-linum-mode 1)
+
+    ;; use a lambda to generate the linum formatting
+    ;; this will use the minimum number of columns,
+    ;; but also always right-align
+    (setq linum-format (lambda (line)
+                         (let* ((w (length (number-to-string (count-lines (point-min) (point-max)))))
+                                (thing (concat " %" (number-to-string (max 2 w)) "d ")))
+                           (propertize (format thing line) 'face 'linum))))
+
+    ;; disable linum in certain modes
+    (require 'linum-off)
+    (add-to-list 'linum-disabled-modes-list 'package-menu-mode)
+    (add-to-list 'linum-disabled-modes-list 'magit--mode)))
+
+(defun kotct/display-line-numbers--turn-off ()
+  "Turns off `display-line-numbers-mode' for the current buffer."
+  (display-line-numbers-mode -1))
```

Comment:
... why? `(kotct/display-line-numbers--turn-off)` is literally more typing than `(display-line-numbers-mode -1)`

Reply:
Oh I see, it's for the hooks. Comments there.

Reply:
Yes, that's exactly why.